src/makeCalculator.js

```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    add(number) {
+      this.result += number;
+
+      return this;
+    },
+    subtract(number) {
+      this.result -= number;
+
+      return this;
+    },
+    multiply(number) {
+      this.result *= number;
+
+      return this;
+    },
+    divide(number) {
+      this.result /= number;
+
+      return this;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(action, number) {
+      action.call(this, number);
+
+      return this;
```

Comment:
```suggestion
      return action.call(this, number);
```

Action already returns `this`.